### ISSUE TITLE
Somewhat inconsistent router.current.uri in nested redirects

### ISSUE THREAD

**Issue by andviro** [gabrielmoreira/riot-router#5]
Hello! First of all, cudos and big thanks for your work on this library! Finally I can ditch React and use Riot for complex applications. I think it deserves (and needs) huge promotion.

I spotted a little inconsistency in current URI reporting, though. 
Consider some nested redirect, for example from `''` to `'/docs'` then from `''` to `'/incoming'` (in `docs` nested route). After application of all redirects, `router.current.uri` equals to `'docs/incoming'`. Somewhere along the way, leading `'/'` was stripped away. But if I click on some `href` pointing to `'#/docs/incoming'`, resulting `router.current.uri` will be `'/docs/incoming'`, _with_ leading slash. It's not a big deal and easily taken into consideration, but nevertheless a bit inconvenient.

**Comment by gabrielmoreira** [comment]
Thanks @andviro.

Yes, i know there are some inconsistences... I don't like how router.current.uri and respose are build. When i added the middleware feature, i realized some problems with redirections, nested routes, etc... 

I think it's a problem that need to be solved with some architectural changes...

Anyway, do you have some sugestion?

**Comment by andviro** [comment]
Hi, Gabriel! I'm sorry, currently too owerwhelmed with current in-house project, can't think about anything else. For now I've decided to go with https://github.com/TehShrike/abstract-state-router, which has additional bonus of being independent from view layer. That said, I still like your general router approach better. When all routes are in one place it reduces cognitive load IMO. Closing this for now, in hope to find some time to participate.
